flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/tableImpl.scala

```diff
@@ -451,6 +450,117 @@ class TableImpl(
     }
     tableName
   }
+
+  override def addColumns(fields: String): Table = {
+    addColumns(ExpressionParser.parseExpressionList(fields): _*);
+  }
+
+  override def addColumns(fields: Expression*): Table = {
+    addColumns(false, fields: _*)
+  }
+
+  override def addColumns(replaceIfExist: Boolean, fields: String): Table = {
+    addColumns(replaceIfExist, ExpressionParser.parseExpressionList(fields): _*)
+  }
+
+  override def addColumns(replaceIfExist: Boolean, fields: Expression*): Table = {
+
+    val aggNames = extractAggregationsAndProperties(
+      fields.map(expressionBridge.bridge), tableEnv)._1
+
+    if(aggNames.nonEmpty){
+      throw new TableException(
+        s"The added field expression cannot be an aggregations, find [${aggNames.head}].")
+    }
+
+    val childFields = logicalPlan.output.map(a => UnresolvedFieldReference(a.name))
+
+    if (replaceIfExist) {
+
+      val finalFields = new ListBuffer[Expression]()
+      val addFields = fields.map(expressionBridge.bridge)
+      childFields.foreach(e => finalFields.append(e))
+
+      // replace field if exist.
+      addFields.foreach {
+        case e@Alias(_, name, _) =>
+          val index = finalFields.indexWhere(p => p match {
+            case u: UnresolvedFieldReference => u.name.equalsIgnoreCase(name)
+            case a: Alias => a.name.equalsIgnoreCase(name)
+            case _ => false
+          })
+          if (index >= 0) {
+            finalFields(index) = e
```

Comment:
Will we allow added fields have the same field name?

Reply:
I think is better to overwrite the field which has the same name. what do you think?